src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -87,33 +87,42 @@
                                           (filter dir-create?)
                                           (map #(.toPath (:path %)))))))
 
+(defn retrieve-events
+  [watcher]
+  (let [watch-key (.take (:watch-service watcher))
+        events (.pollEvents watch-key)]
+    [watch-key events]))
+
+(defn process-events
+  [watcher watch-key java-events shutdown-fn]
+  (let [clojure-events (map #(clojurize % (.watchable watch-key)) java-events)
+        callbacks @(:callbacks watcher)]
+    (log/info (trs "Got {0} event(s) for watched-path {1}"
+                   (count java-events) (.watchable watch-key)))
+    (log/debugf "%s\n%s"
+                (trs "Events:")
+                (pprint-events clojure-events))
+    (log/tracef "%s\n%s"
+                (trs "orig-events:")
```

Comment:
Would be good to settle on orig-events or java-events and just have one. Having both means trying to figure out which clojure var the "orig-events" string is referring to. I'd vote to stick with orig-events, as users seeing logs probably aren't going to care about the java part of the event as much as that it is a preprocessed event.